lib/knapsack_pro/batch.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module KnapsackPro
+  class Batch
+    NotExecutedError = Class.new(StandardError)
+
+    attr_reader :test_file_paths
+
+    def initialize(test_file_paths)
+      @test_file_paths = test_file_paths
+      @passed = nil
+    end
+
+    def executed?
+      !@passed.nil?
+    end
+
+    def passed?
+      raise NotExecutedError.new unless executed?
+      return @passed
+    end
+
+    def _passed
+      @passed = true
+    end
+
+    def _failed
+      @passed = false
+    end
+  end
```

Comment:
I'd propose:

```suggestion
  class Batch
    attr_reader :test_file_paths

    def initialize(test_file_paths)
      @test_file_paths = test_file_paths
      @status = :not_executed
    end

    def executed?
      @status != :not_executed
    end

    def passed?
      @status == :passed
    end

    def _passed
      @status = :passed
    end

    def _failed
      @status = :failed
    end
  end
```

Reply:
Actually, I'd just expose `@status` and avoid `executed?` and `passed?`. I thinks it simplifies even more.

Reply:
Ok. I've applied changes.